routes/api/contacts.js

```diff
@@ -104,9 +104,28 @@ router.put("/:contactId", async (req, res, next) => {
     if (error.message === "Contact not found") {
       res.status(404).json({ message: "Not found" });
     } else {
-        console.error(error);
-        res.status(500).json({ error: "Unknow error occured." })
-        }
+      console.error(error);
+      res.status(500).json({ error: "Unknow error occured." });
+    }
+  }
+});
+
+router.patch("/:contactId/favorite", async (req, res, next) => {
+  const { contactId } = req.params;
+  try {
+    const addToFavorite = await addContactToFavorite(contactId);
+    res.status(200).json({
+      status: "success",
+      code: 201,
+      data: { addToFavorite },
+    });
+  } catch (error) {
+    if (error.message === "Missing field favorite") {
```

Comment:
To chcemy zrobić jako walidację w joi. Tak samo jak w put i post tylko kolejna schema z polem favorite. (tam wsumie też możesz dać odzielną scheme do put bez required())